### DISCUSSION TITLE
Electricity price smartenergy

### DISCUSSION THREAD

**Discussion by MrJamsn** [mampfes/ha_epex_spot#145] [General]
First of all, THANK YOU for this awesome piece of work! highly appreciated.
I have the normal API request in HA too and it delivers the correct price without the addon fee.

You wrote:

```
To get the actual, current Net Price [listed by smartENERGY on their website](https://www.smartenergy.at/smartcontrol#:~:text=Aktueller%20Stundenpreis), configure:

Absolute surcharge = ¢1.2
Tax = 20%

```

and this is exactly what I see coming from your integration, so I put +1.2)*1.2 to my lovelace card and it it correct now (same price as on their homepage https://www.smartenergy.at/smartcontrol "Aktueller Stundenpreis"

These are my sensors:

```
sensor:
  - platform: rest
    name: Electricity Price
    resource: https://apis.smartenergy.at/market/v1/price
    method: GET
    scan_interval: 3600  # Update every hour
    value_template: "{{ value_json.data[0].value }}"
    json_attributes:
      - tariff
      - unit
      - interval
      - data
  - platform: template
    sensors:
      current_electricity_price:
        friendly_name: "Current Electricity Price"
        unit_of_measurement: 'ct/kWh'
        value_template: >
          {% set prices = state_attr('sensor.electricity_price', 'data') %}
          {% if prices %}
            {% set now = (as_timestamp(now()) | timestamp_local) %}
            {% set price = prices | selectattr('date', 'le', now) | list | last %}
            {% if price %}
              {{ price.value + 1.44 }}
            {% else %}
              No data
            {% endif %}
          {% else %}
            No data
          {% endif %}
```

          
          Do I have an error in my thinking/logic or do you use a different API? :)

**Comment by mampfes** [comment]
Hi, the problem is that some providers already include tax, surcharges - and others don't. Therefore you configure it to your needs, your contract with the service provider. If the numbers are correct, then everything is fine, or?

**Comment by Presl1983** [comment]
same here.

it would be cool to get it configurable for epex!
